dff/messengers/telegram.py

```diff
@@ -0,0 +1,179 @@
+                    if attachment_bytes is not None:
+                        files += [InputMediaPhoto(attachment_bytes)]
+                if isinstance(attachment, Document):
+                    attachment_bytes = await attachment.get_bytes(self)
+                    if attachment_bytes is not None:
+                        files += [InputMediaDocument(attachment_bytes)]
+                if isinstance(attachment, Keyboard):
+                    buttons = attachment.buttons
+            if len(files) > 0:
+                await bot.send_media_group(chat_id, files, caption=message.text)
+                return
+        if message.text is not None:
+            await bot.send_message(chat_id, message.text, reply_markup=self._create_keyboard(buttons))
+
+    async def _on_event(self, update: Update, _: ContextTypes.DEFAULT_TYPE, create_message: Callable[[Update], Message]) -> None:
+        if update.effective_chat is not None and update.message is not None:
+            message = create_message(update)
+            message.original_message = update
+            resp = await self.callback(message, update.effective_chat.id)
+            if resp.last_response is not None:
+                await self.cast_message_to_telegram_and_send(self.application.bot, update.effective_chat.id, resp.last_response)
+
+    async def on_message(self, update: Update, _: ContextTypes.DEFAULT_TYPE) -> None:
+        await self._on_event(update, _, lambda u: await self.extract_message_from_telegram(u.message))
+
+    async def on_callback(self, update: Update, _: ContextTypes.DEFAULT_TYPE) -> None:
+        await self._on_event(update, _, lambda u: Message(attachments=[CallbackQuery(query_string=u.callback_query.data)]))
+
+    async def connect(self, callback: PipelineRunnerFunction, *args, **kwargs):
+        self.callback = callback
+
+
+class PollingTelegramInterface(_AbstractTelegramInterface):  # pragma: no cover
+    def __init__(self, token: str, interval: int = 2, timeout: int = 20) -> None:
+        super().__init__(token)
+        self.interval = interval
+        self.timeout = timeout
+
+    async def connect(self, callback: PipelineRunnerFunction, *args, **kwargs):
+        await super().connect(callback, *args, **kwargs)
+        self.application.run_polling(poll_interval=self.interval, timeout=self.timeout, allowed_updates=Update.ALL_TYPES)
+
+
+class CallbackTelegramInterface(_AbstractTelegramInterface):  # pragma: no cover
+    def __init__(self, token: str, host: str = "localhost", port: int = 844):
+        super().__init__(token)
+        self.listen = host
+        self.port = port
+    
+    async def connect(self, callback: PipelineRunnerFunction, *args, **kwargs):
+        await super().connect(callback, *args, **kwargs)
+        self.application.run_webhook(listen=self.listen, port=self.port, allowed_updates=Update.ALL_TYPES)
+
+
+def telegram_condition(func: Callable[[Update], bool]):  # pragma: no cover
```

Comment:
We don't need a telegram condition anymore.
We could just tell users to write custom conditions that check the `original_message` attribute.

Reply:
Checking `last_request is None` is unnecessary.

`original_message` casting is nice but I think it might be better to add a method that casts it (similar to `populate_attachment`).